File: AndroidBeginner/CheckingConnection/network-connect-2/src/main/java/algokelvin/app/networkconnect/ConnectionController.java
package algokelvin.app.networkconnect;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import androidx.appcompat.app.AppCompatActivity;

public class ConnectionController extends AppCompatActivity {
    protected boolean isConnected = false;

    protected void isNetworkConnected() {
        ConnectivityManager cm = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = cm.getActiveNetworkInfo();
        isConnected = activeNetworkInfo != null && activeNetworkInfo.isConnected();
    }
}

File: AndroidBeginner/CheckingConnection/network-connect-3/src/main/java/algokelvin/app/network_connect/ConnectionController.kt
package algokelvin.app.network_connect

import android.net.ConnectivityManager
import androidx.appcompat.app.AppCompatActivity

open class ConnectionController: AppCompatActivity() {
    protected var isConnected = false

    protected fun isNetworkConnected() {
        val cm = getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetworkInfo = cm.activeNetworkInfo
        isConnected = activeNetworkInfo != null && activeNetworkInfo.isConnected
    }
}